Purchasing.Core/Services/AggieEnterpriseService.cs

```diff
@@ -1,17 +1,352 @@
+                        {
+                            Percent = dist.Value,
+                            GlSegmentString = dist.Key.FinincialSegmentString,
+                        });
+                    }
+                }
+
+                li.Distributions = aeDist.ToArray();
+
+
+                lineItems.Add(li);
+            }
+            
+
+            inputOrder.Payload.Lines = lineItems;
+            
+            var jsonPayload = System.Text.Json.JsonSerializer.Serialize(inputOrder);
+            var log = Log.ForContext("jsonPayload", jsonPayload);
+            log.Information("Aggie Enterprise Payload");
+
+            var NewOrderRequsition = await _aggieClient.ScmPurchaseRequisitionCreate.ExecuteAsync(inputOrder);
+            
+            var responseData = NewOrderRequsition.ReadData();
+
+            log.Information("Aggie Enterprise Response {response}", System.Text.Json.JsonSerializer.Serialize(responseData));
+            var rtValue = new SubmitResult { Success = false, Messages = new List<string>() };
+
+            if (responseData.ScmPurchaseRequisitionCreate.RequestStatus.RequestStatus == RequestStatus.Rejected || responseData.ScmPurchaseRequisitionCreate.RequestStatus.RequestStatus == RequestStatus.Error)
+            {
+                
+                if (responseData.ScmPurchaseRequisitionCreate.RequestStatus.Equals(RequestStatus.Error))
+                {
+                    rtValue.Messages.Add("Aggie Enterprise Error");
+                    foreach(var message in responseData.ScmPurchaseRequisitionCreate.RequestStatus.ErrorMessages)
+                    {
+                        rtValue.Messages.Add($"Error: {message}");
+                    }
+                }
+                else
+                {
+                    rtValue.Messages.Add("Aggie Enterprise Rejected");
+                    foreach (var message in responseData.ScmPurchaseRequisitionCreate.ValidationResults.ErrorMessages)
+                    {
+                        rtValue.Messages.Add(message);
+                    }
+                }
+
+                return rtValue;
+            }
+
+            rtValue = new SubmitResult { Success = true, DocNumber = responseData.ScmPurchaseRequisitionCreate.RequestStatus.RequestId.ToString() };
+
+            return rtValue;
+        }
+
+        public async Task<IScmPurchaseRequisitionRequestStatus_ScmPurchaseRequisitionRequestStatus_RequestStatus> LookupOrderStatus(string requestId)
+        {
+            try
+            {
+                var result = await _aggieClient.ScmPurchaseRequisitionRequestStatus.ExecuteAsync(requestId);
+
+                var data = result.ReadData();
+
+                return data.ScmPurchaseRequisitionRequestStatus.RequestStatus;
+            }
+            catch
+            {
+                Log.Warning("Aggie Enterprise LookupOrderStatus failed for {requestId}", requestId);
+                return null;
```

Comment:
This shows a "can't connect" type message to the user.
I think the only way this would happen is if AE is down, or the change the request id manually and it is the wrong format or not there.